server/routes/gcp/auth.py

```
@@ -224,6 +224,228 @@ def force_disconnect_gcp(user_id):
+    try:
+        listed = get_project_list(creds)
+        accessible_projects = [
+            {"project_id": p.get("projectId"), "name": p.get("name") or p.get("projectId")}
+            for p in listed
+            if p.get("projectId")
+        ] or home_fallback
+        if accessible_projects is home_fallback:
+            logging.warning(
+                "GCP SA connect: projects.list returned empty — falling back to SA's embedded project_id"
+            )
+    except Exception as exc:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)